server/src/main/java/com/paypal/selion/grid/servlets/transfer/DefaultCriteria.java

```diff
@@ -0,0 +1,118 @@
+/*-------------------------------------------------------------------------------------------------------------------*\
+|  Copyright (C) 2015 PayPal                                                                                          |
+|                                                                                                                     |
+|  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance     |
+|  with the License.                                                                                                  |
+|                                                                                                                     |
+|  You may obtain a copy of the License at                                                                            |
+|                                                                                                                     |
+|       http://www.apache.org/licenses/LICENSE-2.0                                                                    |
+|                                                                                                                     |
+|  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed   |
+|  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for  |
+|  the specific language governing permissions and limitations under the License.                                     |
+\*-------------------------------------------------------------------------------------------------------------------*/
+
+package com.paypal.selion.grid.servlets.transfer;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.commons.lang.StringUtils;
+
+import com.paypal.selion.grid.servlets.transfer.DefaultManagedArtifact.DefaultRequestParameters;
+import com.paypal.selion.logging.SeLionGridLogger;
+
+/**
+ * {@link Criteria} to match a {@link DefaultManagedArtifact} uniquely. Criteria uses artifact name, user id and
+ * artifact folder to uniquely identify a {@link DefaultManagedArtifact}. Parameters artifactName, artifactFolder and
+ * userId match artifact name, folder name and parent folder name of some {@link DefaultManagedArtifact}
+ * respectively.
+ */
+final class DefaultCriteria implements Criteria {
+    
+    private static final SeLionGridLogger LOGGER = SeLionGridLogger.getLogger(DefaultCriteria.class);
+
+    private String artifactName;
+
+    private String uuidFolderName;
+
+    private String subFolderName;
+
+    public DefaultCriteria(Map<String, String> parametersMap) {
+        validateParametersMap(parametersMap);
+        this.artifactName = parametersMap.get(ManagedArtifact.ARTIFACT_FILE_NAME);
+        this.uuidFolderName = parametersMap.get(DefaultRequestParameters.USER_ID);
```

Comment:
Well, it's really a "uid".. Nothing about the intent requires the identifier to be associated with a user. It just needs to be a unique identifier. Probably the request parameter should be changed to reflect this too. I left that part in tact, as you can see.